File: language/python/week2/1541.py
print("수식을 입력하세요.")
expression = list(input())
# print(expression)
number_temp = 0
sum = 0
n = 1
for i in range(len(expression)):
    number_temp += 1
    if expression[i] >= '0' and expression[i] <= '9' and n == 1:
        if number_temp > 1:
            sum -= int(expression[i-1])
            sum = sum + ((int(expression[i-1])*10 + int(expression[i])))
        else:
            sum += int(expression[i])
    elif expression[i] >= '0' and expression[i] <= '9' and n == -1:
        if number_temp > 1:
            sum += int(expression[i-1])
            sum = sum - ((int(expression[i-1])*10 + int(expression[i])))
        else:
            sum -= int(expression[i])
    else:
        number_temp = 0
        if expression[i] == '+':
            continue
        elif expression[i] == '-':
            n = -1

print("sum : ", sum)

File: language/c/week2/1541.c
//
//  main.c
//  baekjun
//
//  Created by 이지원 on 2021/09/06.
//

#include <stdio.h>
#include <string.h>

int main(int argc, const char *argv[])
{
    // insert code here...
    char str[50];
    int n = 1;
    int number_temp = 0;
    int sum = 0;
    printf("수식을 입력하세요.\n");
    scanf("%s", str);

    for (int i = 0; i < strlen(str); i++)
    {
        number_temp += 1;
        if ('0' <= str[i] && '9' >= str[i] && n == 1)
        {
            if (number_temp > 1)
            {
                sum -= str[i - 1] - '0';
                sum = sum + ((str[i - 1] - '0') * 10 + str[i] - '0');
            }
            else
            {
                sum += str[i] - '0';
            }
        }
        else if ('0' <= str[i] && '9' >= str[i] && n == -1)
        {
            if (number_temp > 1)
            {
                sum += str[i - 1] - '0';
                sum = sum - ((str[i - 1] - '0') * 10 + str[i] - '0');
            }
            else
            {
                sum -= str[i] - '0';
            }
        }
        else
        {
            number_temp = 0;
            if (str[i] == '+')
            {
                continue;
            }
            else if (str[i] == '-')
            {
                n = -1;
            }
        }
    }
    printf("sum : %d\n", sum);

    return 0;
}
